### ISSUE TITLE
Add make_loader and make_dumper functions

### ISSUE THREAD

**Issue by LourensVeen** [yatiml/yatiml#25]
The current way of making a Loader or Dumper by defining a class is clunky, and a single function call would be nicer. I just didn't know how to implement it, but I think I do now. So let's have

```
MyLoader = yatiml.make_loader(TopClass, [OtherClass, OtherClass2])
MyDumper = yatiml.make_dumper(TopClass, OtherClass, Otherclass2, output_format='json')
```
Also, update the documentation to use it everywhere.

**Comment by LourensVeen** [comment]
Actually, we could just return a callable that loads the file, and wraps ruamel.yaml. With its uncertain status in the face of the move to SourceForge and the ruyaml fork, hiding some more of it behind the YAtiML facade may be a good idea.

Maybe we can just  follow the Python `json` API convention?
```
load_config = yatiml.load_function(Config, OtherClass, OtherClass2)
with open('config.yaml', 'r') as f:
    config = load_config(f)

dump_config = yatiml.dump_function(Config, OtherClass, OtherClass2)
with open('new_config.yaml', 'w') as f:
    save_config(config, f)

dump_config_as_json = yatiml.dump_json_function(Config, OtherClass, OtherClass2)
with open('new_config.json', 'w') as f:
    dump_config_as_json(config, f)

dumps_config = yatiml.dumps_function(Config, OtherClass, OtherClass2)
yaml_text = dumps_config(config)

dumps_config_as_json = yatiml.dumps_json_function(Config, OtherClass, OtherClass2)
json_text = dumps_config_as_json(config)
```
And then maybe for convenience we could allow passing a `str` or a `Path` as well? That's what PyYaml does anyway.
```
load_config = yatiml.load_function(Config, OtherClass, OtherClass2)
config = load_config('config.yaml')
```
That looks nice.

**Comment by LourensVeen** [comment]
Released with 0.6.0.
